### PR TITLE
chore: added k8s validation on user inputs

### LINKED ISSUE #15: Validate user inputs (name, images)

### PR DESCRIPTION
**Pull request by mojtaba-esk**
Closes #15 

## Summary by CodeRabbit

- **New Features**
	- Introduced comprehensive validation checks for multiple Kubernetes resource creation methods, enhancing input integrity and error handling.
	- Added a new file that provides a set of validation functions for compliance with naming conventions and structural integrity.
	- Introduced a new file with a comprehensive suite of unit tests for validating Kubernetes naming conventions and configurations.

- **Bug Fixes**
	- Enhanced error handling by introducing new error variables for various validation issues.

- **Tests**
	- Modified test structures to include new configuration options for enhanced testing capabilities of Kubernetes resources.
	- Added a comprehensive suite of tests to validate critical Kubernetes validation logic.

### CODE REVIEW

File: pkg/k8s/validate.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation"
+
+	"github.com/celestiaorg/knuu/pkg/errors"
+)
+
+func validateDNS1123Label(name string, err *errors.Error) error {
+	if errs := validation.IsDNS1123Label(name); len(errs) > 0 {
```
Comment (smuu) [inline pkg/k8s/validate.go:14]: Does this also check the length of the label? In Kubernetes, there is a max limit.

File: pkg/k8s/validate.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+		return err.WithParams(name, errs)
+	}
+	return nil
+}
+
+func validateDNS1123Subdomain(name string, returnErr *errors.Error) error {
+	if errs := validation.IsDNS1123Subdomain(name); len(errs) > 0 {
```
Comment (smuu) [inline pkg/k8s/validate.go:21]: Does this also check the length of the label? In Kubernetes, there is a max limit.

**Review by smuu — APPROVED** [review]
Amazing job!

File: pkg/k8s/validate.go (hunk 1 above)
Comment (mojtaba-esk) [reply pkg/k8s/validate.go:14]: yes it does

File: pkg/k8s/validate.go (hunk 2 above)
Comment (mojtaba-esk) [reply pkg/k8s/validate.go:21]: yes it does

**Pull request merged**